.github/workflows/OCV-Android-SDK.yaml

```diff
@@ -0,0 +1,77 @@
+  workflow_call:
+
+env:
+  PR_AUTHOR: ${{ github.event.pull_request.user.login }}
+  PR_AUTHOR_FORK: ${{ github.event.pull_request.head.repo.full_name }}
+  SOURCE_BRANCH_NAME: ${{ github.head_ref }}
+  TARGET_BRANCH_NAME: ${{ github.base_ref }}
+  GIT_CACHE_DOCKER: '/home/ci/git_cache'
+  OPENCV_DOCKER_WORKDIR: '/opencv'
+
+jobs:
+  BuildAndTest:
+    runs-on: opencv-cn-lin-x86-64
+    defaults:
+      run:
+        shell: bash
+    container:
+      image: quay.io/opencv-ci/androidsdk
+      volumes:
+        - /home/opencv-cn/git_cache:/home/ci/git_cache
+        - /home/opencv-cn/ci_cache/opencv:/home/ci/.ccache
+    steps:
+    - name: Setup infra environment
+      if: ${{ github.event.repository.name == 'ci-gha-workflow' }}
+      run: echo "TARGET_BRANCH_NAME=4.x" >> $GITHUB_ENV
+    - name: PR info
+      run: |
+        echo "PR Author: ${{ env.PR_AUTHOR }}"
+        echo "PR Author fork: ${{ env.PR_AUTHOR_FORK }}"
+        echo "Source branch name: ${{ env.SOURCE_BRANCH_NAME }}"
+        echo "Target branch name: ${{ env.TARGET_BRANCH_NAME }}"
+    - name: Clean
+      run: find ${{ env.OPENCV_DOCKER_WORKDIR }} -mindepth 1 -delete
+    - name: Fetch opencv
+      run: git clone --branch ${{ env.TARGET_BRANCH_NAME }} --reference ${{ env.GIT_CACHE_DOCKER }}/opencv.git https://github.com/opencv/opencv.git ${{ env.OPENCV_DOCKER_WORKDIR }}
+    - name: Merge opencv with ${{ env.SOURCE_BRANCH_NAME }} branch
+      if: ${{ github.event.repository.name != 'ci-gha-workflow' }}
+      run: |
+        cd ${{ env.OPENCV_DOCKER_WORKDIR }}
+        git config --global --add safe.directory ${{ env.OPENCV_DOCKER_WORKDIR }}
+        git config user.email "opencv.ci"
+        git config user.name "opencv.ci"
+        git pull -v "https://github.com/${{ env.PR_AUTHOR_FORK }}" "${{ env.SOURCE_BRANCH_NAME }}"
+
+    - name: Build OpenCV for Android SDK
+      run: |
+        cd /build
+        python3 "${{ env.OPENCV_DOCKER_WORKDIR }}/platforms/android/build_sdk.py" --build_doc --config "${{ env.OPENCV_DOCKER_WORKDIR }}/platforms/android/ndk-18.config.py" --sdk_path "$ANDROID_HOME" --ndk_path "$ANDROID_NDK_HOME" /build
+
+    - name: Test CMake
+      run: cd /build && python3 "${{ env.OPENCV_DOCKER_WORKDIR }}/platforms/android/build-tests/test_cmake_build.py" OpenCV-android-sdk/sdk/native/jni
+
+    - name: Test Ant
+      run: cd /build && python3 "${{ env.OPENCV_DOCKER_WORKDIR }}/platforms/android/build-tests/test_ant_build.py" OpenCV-android-sdk/sdk/java OpenCV-android-sdk/samples
+
+    - name: Test NDK
+      run: cd /build && python3 "${{ env.OPENCV_DOCKER_WORKDIR }}/platforms/android/build-tests/test_ndk_build.py" OpenCV-android-sdk/sdk/native/jni
+
+    - name: Test Gradle
+      run: cd /build && python3 "${{ env.OPENCV_DOCKER_WORKDIR }}/platforms/android/build-tests/test_gradle.sh" OpenCV-android-sdk
+
+    #- name: Create Package
+      #run: cd build && zip -r -9, -y OpenCV4Android.zip OpenCV-android-sdk
```

Comment:
How we want to add artifacts to release? Do they should be on a build page or inside a release? I guess it should be inside of a release, but in this case we have two ways how to do it:

1. Creating a release w/o artifacts which will trigger this job and upload artifacts
`if: github.event_name == 'release'`
2. We will create a release by a GHA job which will be done after building all artifacts. It should run on push to `v*`

@asmorkalov @alalek what do you think?

Reply:
Actually we usually prepare packages for all nightly builds. At least to verify (and fix) them **before** the release.
No idea how to deal with GHA.